au/code/au/utility/type_traits.hh

```diff
@@ -39,11 +44,29 @@ struct AlwaysFalse : std::false_type {};
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 // Implementation details below.
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// `Prepend` implementation.
```

Comment:
What is the point of this comment? Do we need a comment for all inline implementations?

Reply:
Often, the implementation for "public"-ish interface declared up top in a file will depend on new helpers, which I will declare-and-define right next to the "core" implementation.  This can get a little messy and confusing, so I use these "section comments" to group things and clarify the purpose.  That said, I'm certainly rather inconsistent about using them in practice.